packages/asset-swapper/src/quote_consumers/coordinator_swap_quote_consumer.ts

```diff
@@ -0,0 +1,231 @@
+                coordinatorSignatures,
+                { from: signedZeroExTransaction.signerAddress },
+            );
+            return txHash;
+        } catch (err) {
+            if (_.includes(err.message, ContractError.SignatureRequestDenied)) {
+                throw new Error(SwapQuoteConsumerError.SignatureRequestDenied);
+            } else {
+                throw err;
+            }
+        }
+    }
+    private async _getCoordinatorZeroExTransactionAsync(
+        quote: SwapQuote,
+        opts: Partial<CoordinatorSwapQuoteGetOutputOptsBase>,
+    ): Promise<SignedZeroExTransaction> {
+        assert.isValidSwapQuote('quote', quote);
+        const takerAddress = opts.takerAddress as string;
+        assert.isETHAddressHex('opts.takerAddress', takerAddress);
+
+        const { params } = await this.getSmartContractParamsOrThrowAsync(quote, opts);
+
+        const { orders, signatures } = params;
+        let zeroExOperation: string;
+        if (params.type === MarketOperation.Buy) {
+            const { makerAssetFillAmount } = params;
+            zeroExOperation = this._contractWrappers.exchange.marketBuyOrdersNoThrow.getABIEncodedTransactionData(
+                orders,
+                makerAssetFillAmount,
+                signatures,
+            );
+        } else {
+            const { takerAssetFillAmount } = params;
+            zeroExOperation = this._contractWrappers.exchange.marketSellOrdersNoThrow.getABIEncodedTransactionData(
+                orders,
+                takerAssetFillAmount,
+                signatures,
+            );
+        }
+        const transaction: ZeroExTransaction = {
+            salt: generatePseudoRandomSalt(),
+            signerAddress: takerAddress,
+            data: zeroExOperation,
+            verifyingContractAddress: this._contractWrappers.exchange.address,
+        };
+        // If the takerAddress is a contract we need to generate an `isValidSignature` SignatureType and skip
+        // the signing via the provider.
+        const contractCode = await this._web3Wrapper.getContractCodeAsync(takerAddress);
+        if (contractCode === NULL_BYTES || contractCode === '0x00') {
+            // Sign as an EOA user
+            const signedZeroExTransaction = await signatureUtils.ecSignTransactionAsync(
+                this.provider,
+                transaction,
+                takerAddress,
+            );
+            return signedZeroExTransaction;
+        } else {
+            /*
+                function isValidSignature(
+                        bytes32, // hash
+                        bytes calldata // signature
+                )
+                    external
+                    pure
+                    returns (bytes4)
+                {
+                    return IS_VALID_WALLET_SIGNATURE_MAGIC_VALUE;
+                }
+             */
+            const signature = signatureUtils.convertToSignatureWithType(NULL_BYTES, SignatureType.Wallet);
```

Comment:
TODO Handle the case where the contract does not have a general "Approve All" signature.